### PR TITLE
dsa-messaging

### PR DESCRIPTION
**Pull request by darmhoo**

### CODE REVIEW

File: src/views/admin/Dashboard.vue [hunk 1]
```diff
@@ -227,16 +227,25 @@
...
           </div>
           <div
             v-on:click="resetSearch()"
-            class="w-2/5 py-2 altaraBlue rounded h-10 text-white text-center"
+            class="w-1/6 py-2 altaraBlue rounded h-10 text-white text-center"
           >
             Reset
           </div>
+
+          <div
```
Comment (teebarg) [inline src/views/admin/Dashboard.vue:242]: let put this behind the flag

File: src/views/admin/SendMessage.vue [hunk 2]
```diff
@@ -1,131 +1,148 @@
...
+    this.fetchDsa();
+  },
+  methods: {
+    async fetchDsa() {
+      this.$LIPS(true);
+      const agents = await get(this.URLS.roles);
+      this.roles = agents.data.roles;
+      this.$LIPS(false);
+    },
+    addToArray() {
```
Comment (teebarg) [inline src/views/admin/SendMessage.vue:91]: is this still relevant??

File: src/views/admin/SendMessage.vue [hunk 3]
```diff
@@ -1,131 +1,148 @@
...
+      if (this.recepients.includes(this.selectedUser)) {
+        return;
+      }
+      this.recepients.push(this.selectedUser);
+    },
+    removeFromArray(item) {
```
Comment (teebarg) [inline src/views/admin/SendMessage.vue:97]: is vue multiselect not handling this??

File: src/views/admin/SendMessage.vue [hunk 4]
```diff
@@ -1,131 +1,148 @@
...
-                    roles : ids,
-                    message: this.message
-                }                    
-                    post(this.URLS.sendMessage, data).then(resp => {
-                        console.log(resp);
+      this.recepients = this.recepients.filter((a) => {
+        return a.id !== item.id;
+      });
+    },
+    addAll() {
```
Comment (teebarg) [inline src/views/admin/SendMessage.vue:104]: are these not your custom stuff?

File: src/views/admin/SendMessage.vue [hunk 5]
```diff
@@ -1,131 +1,148 @@
...
+        let ids = this.recepients.map((item) => {
+          return item.id;
+        });
+        let data = {
+          roles: ids,
+          message: this.message,
+        };
+        post(this.URLS.sendMessage, data)
+          .then((resp) => {
+            console.log(resp);
```
Comment (teebarg) [inline src/views/admin/SendMessage.vue:122]: remove logs

File: src/views/admin/Dashboard.vue (hunk 1 above)
Comment (darmhoo) [reply src/views/admin/Dashboard.vue:242]: done

File: src/views/admin/SendMessage.vue (hunk 2 above)
Comment (darmhoo) [reply src/views/admin/SendMessage.vue:91]: done

File: src/views/admin/SendMessage.vue (hunk 4 above)
Comment (darmhoo) [reply src/views/admin/SendMessage.vue:104]: yes

**Review by teebarg — APPROVED** [review]

**Pull request merged**